x-pack/plugins/files/server/file_share_service/internal_file_share_service.ts

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import moment from 'moment';
+import type {
+  SavedObjectsClientContract,
+  SavedObject,
+  ISavedObjectsRepository,
+} from '@kbn/core/server';
+import type {
+  FileSavedObjectAttributes,
+  FileShareJSON,
+  FileShareSavedObjectAttributes,
+  UpdatableFileShareAttributes,
+} from '../../common/types';
+import { FILE_SO_TYPE } from '../../common/constants';
+import type { File } from '../file';
+import { fileShareObjectType } from '../saved_objects';
+import { generateShareToken } from './generate_share_token';
+import { FileShareServiceStart } from './types';
+
+export interface CreateShareArgs {
+  /**
+   * Optionally provide a name for this file share instance
+   */
+  name?: string;
+  /**
+   * Optionally set an expiration date for this file share instance
+   *
+   * @note If not specified the file share will expire after 30 days
+   */
+  validUntil?: string;
```

Comment:
I'm thinking maybe we can make this a `number` a UNIX timestamp or JavaScript timestamp. This way we don't need to parse it with `moment` library, which is a potential for attacks.